### PR TITLE
Feature/raw response

### PR DESCRIPTION
**Pull request by Essk**
## Description
As discussed at https://github.com/elastic/search-ui/pull/439#discussion_r346568681
This PR adds a `rawResponse` field to the state set up by the site-search and app-search API connectors.

## List of changes
* Add `rawResponse` field to ite-search and app-search API connectors
* Update test data to include raw rsponse field
* Update documentation

## Associated Github Issues
Addresses #436

### CODE REVIEW

**Comment by JasonStoltz** [comment]
@Essk Thank you for the PR, reviewing now.

File: packages/search-ui-app-search-connector/src/__tests__/responseAdapter.test.js [hunk 1]
```diff
@@ -189,10 +251,113 @@ const adaptedResponse = {
...
+        request_id: "1234",
+      },
+    },
+    rawResults: [],
+  },
+  requestId: "1234",
+  results: []
+};
+
+const adaptedResponseWithEmptyFacetValue = {
```
Comment (JasonStoltz) [inline packages/search-ui-app-search-connector/src/__tests__/responseAdapter.test.js]: I merged master into this PR from the Github UI and it looks like we now have duplicate declarations for `adaptedResponseWithEmptyFacetValue`. Could you please resolve this manually? Sorry about that.

File: ADVANCED.md [hunk 2]
```diff
@@ -258,6 +258,7 @@ It is updated indirectly by invoking an action which results in a new API reques
...
 | `autocompletedSuggestionsRequestId` | String                                                                                 | A unique ID for the current autocompleted suggestion results.                                                                                                                                                                                                             |
 | `facets`                            | Object[[Facet](./packages/react-search-ui/src/types/Facet.js)]                         | Will be populated if `facets` configured in [Advanced Configuration](#advanced-configuration).                                                                                                                                                                            |
+| `rawResponse`         | Object[Response] | The response object received from the API |
```
Comment (JasonStoltz) [inline ADVANCED.md]: Could you please add `rawResponse` to `DEFAULT_STATE` in `SearchDriver`? https://github.com/elastic/search-ui/blob/master/packages/search-ui/src/SearchDriver.js#L29

**Comment by davidsauntson** [comment]
@JasonStoltz @Essk do you have any plans to surface the spelling corrections in a different way? I'd be happy to try and pick this up if someone can tell me the outstanding tasks?  We really need to have access to the spelling suggestions.

**Comment by JasonStoltz** [comment]
I saw you opened up a new PR, will move the conversation there.

**Pull request closed without merging**